cleanlab/rank.py

```diff
@@ -504,9 +504,9 @@ def get_self_confidence_for_each_label(
       Lower scores indicate more likely mislabeled examples.
     """
 
-    # To make this work for multi-label (but it will slow down runtime), replace:
-    # pred_probs[i, l] -> np.mean(pred_probs[i, l])
-    return np.array([pred_probs[i, l] for i, l in enumerate(labels)])
+    # To make this work for multi-label (but it will slow down runtime), return:
+    # np.mean(pred_probs[np.arange(labels.shape[0]), labels], axis=1)
```

Comment:
This won't work because `labels: List[List[int]]`.

It won't have a `shape` attribute and won't be a valid index.

Reply:
Gotcha. One possible alternative is - 
```
labels = np.array(labels) # Convert List[List[int]] to np.ndarray
np.mean(pred_probs[np.arange(labels.shape[0]), labels], axis=1)
```
However, if `labels: List[List[int]]` is to be supported, then we might want to specify the parameter type as 
```
labels: Union(List[List[int]], np.ndarray)
```
in the future.

Reply:
Edit: Deleted my previous comment because I didn't understand the issue properly.
What would you suggest here? I am not seeing a vectorized version.

Reply:
Let's just stick to the working version.

```suggestion
    # np.array([np.mean(pred_probs[i, l]) for i, l in enumerate(labels)])
```